pkg/kubelet/dockertools/container_gc.go

```diff
@@ -245,35 +261,47 @@ func (cgc *containerGC) GarbageCollect(gcPolicy kubecontainer.ContainerGCPolicy,
 	return nil
 }
 
-func (cgc *containerGC) removeContainer(id string, podNameWithNamespace string, containerName string) {
-	glog.V(4).Infof("Removing container %q name %q", id, containerName)
-	err := cgc.client.RemoveContainer(id, dockertypes.ContainerRemoveOptions{RemoveVolumes: true})
+func (cgc *containerGC) removeContainer(containerInfo containerGCInfo) {
+	if !containerInfo.isHostNetwork && containerInfo.containerName == leaky.PodInfraContainerName {
```

Comment:
s/leaky.PodInfraContainerName/PodInfraContainerName

It's already defined in `dockertools/docker.go`